frameworks/helloworld/src/test/java/com/mesosphere/sdk/helloworld/scheduler/ServiceTest.java

```diff
@@ -50,6 +52,133 @@ public void testDefaultDeployment() throws Exception {
         runDefaultDeployment();
     }
 
+    /**
+     * Checks that if an unessential task in a pod fails, that the other task in the same pod is unaffected.
+     */
+    @Test
+    public void testNonEssentialTaskFailure() throws Exception {
```

Comment:
NIT:  I've stopped prefixing all tests with `test`.  Things are verbose enough.  It says `@Test` right there.